modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -202,18 +203,18 @@ sealed abstract class Json extends Product with Serializable {
    * The Play docs, from which this method was inspired, reads:
    *   "Lookup for fieldName in the current object and all descendants."
    */
-  final def findAllByKey(key: String): List[Json] = keyValues(this).collect {
-    case (k, v) if (k == key) => v
-  }
-
-  private def keyValues(json: Json): List[(String, Json)] = json match {
-    case JObject(obj)  => obj.toList.flatMap { case (k, v) => keyValuesHelper(k, v) }
-    case JArray(elems) => elems.toList.flatMap(keyValues)
-    case _             => Nil
+  final def findAllByKey(key: String): List[Json] = {
+    val hh: ListBuffer[Json] = ListBuffer.empty[Json]
+    def loop(json: Json): Unit = json match {
+      case JObject(obj)  => obj.toIterable.foreach { case (k, v) =>
+        if (k == key) hh += v
+        loop(v)
+      }
+      case JArray(elems) => elems.foreach(loop)
+      case _             => // do nothing
+    }
```

Comment:
Looks like you need to call `loop(key)` here.

Reply:
Right. It needs the initial call.